tez-dag/src/main/java/org/apache/tez/dag/app/dag/impl/DAGImpl.java

```diff
@@ -2579,11 +2586,16 @@ void stopVertexServices() {
   }
 
   @Override
-  public void addUsedContainer(ContainerId containerId) {
-    containersUsedByCurrentDAG.add(containerId);
+  public void addUsedContainer(Container container) {
+    containersUsedByCurrentDAG.add(container.getId());
+    nodesUsedByCurrentDAG.add(container.getNodeId());
+    nodeHostsUsedByCurrentDAG.add(container.getNodeId().getHost());
```

Comment:
can ``container.getNodeId()`` be ``null`` in any scenario?

Reply:
hm, should not be, in that case it's a yarn bug

Reply:
Ohhk